x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/feature/FeatureFactory.java

```diff
@@ -332,8 +340,11 @@ private MvtCoordinateSequenceFilter(Envelope tileEnvelope, int extent) {
 
         @Override
         public void filter(CoordinateSequence seq, int i) {
-            seq.setOrdinate(i, 0, lon(seq.getOrdinate(i, 0)));
-            seq.setOrdinate(i, 1, lat(seq.getOrdinate(i, 1)));
+            if (i != 0 && i == seq.size() - 1 && seq.getCoordinate(0) == seq.getCoordinate(i)) {
```

Comment:
I think a comment here might be useful. It is especially not clear why we need to do reference equality vs contents equality. We could add a comment like:
```
// JTS can sometimes close polygons using references to the starting coordinate, which can lead to that coordinate being converted twice.
// We need to detect if the end is the same actual Coordinate, and not do the conversion again.
```